brave/brave-core
# Add additional buckets to ethereum provider question, add solana provider question

Resolves https://github.com/brave/brave-browser/issues/29193


Resolves 



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

Follow test plan for both `Brave.Wallet.EthProvider.3` and `Brave.Wallet.SolProvider`. Ensure metric responses are independent of each other.

0) Start with fresh profile. Access brave.com. Ensure metric has a value of 0.
1) Setup the wallet. Reload the brave.com page. Ensure metric has value of 2.
2) Access web3 settings and set the default provider to "none". Reload the page. Ensure the metric has a value of 1.
3) Set the default provider setting to "brave wallet". Reload page. Ensure the metric has a value of 3.
4) Set the default provider setting to "extensions". Install & setup a third-party extension for the given chain. Reload page. Ensure the metric has a value of 5.
5) Set the default provider setting to "none". Reload page. Ensure the metric has a value of 4.
6) Set the default provider setting to "extensions", reset Brave Wallet. Reload page. Ensure metric has a value of 4.
7) Uninstall the third-party extension. Reload page. Ensure the metric has a value of 0.
